umap/static/umap/js/modules/ui/bar.js

```diff
@@ -31,7 +31,8 @@ const TOP_BAR_TEMPLATE = `
         <button class="edit-save button round" type="button" data-ref="save">
             <i class="icon icon-16 icon-save"></i>
             <i class="icon icon-16 icon-save-disabled"></i>
-            <span class="">${translate('Save')}</span>
+            <span data-ref="saveLabel">${translate('Save')}</span>
+            <span data-ref="saveDraftLabel">${translate('Save draft')}</span>
```

Comment:
Maybe putting the `hidden` attribute by default would avoid flashing buttons? Or is that rendered after the `redraw()` anyway?

Reply:
I'm not sure it would blink as `redraw` will always change both `hidden` in the same time with opposite values, but it does not hurt to make both hidden by default, so I've done that!